core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -88,6 +88,11 @@ class TopicConfigHandler(private val logManager: LogManager, kafkaConfig: KafkaC
     if (Try(topicConfig.getProperty(KafkaConfig.UncleanLeaderElectionEnableProp).toBoolean).getOrElse(false)) {
       kafkaController.enableTopicUncleanLeaderElection(topic)
     }
+
+    Try(topicConfig.getProperty(KafkaConfig.MinInSyncReplicasProp).toInt) match {
```

Comment:
It seems if the brokers are already running, and a new topic is created, this code path won't be triggered.
It's only triggered when the config of an existing topic's config is being changed.

Then it'd be a problem for the controller's topicMinIsrConfig field since it won't contain data of the newly created topics. Maybe we'll need to query zk for topic configs upon controller failover and new topic creation.

Reply:
Dang, you're right. That seems like a problem in this DynamicConfigManager thing. I just tested this and you're right, this code path is only exercised on first boot and when configuration is altered.

I think ideally the ConfigHandler would be invoked for topic creation as well.

Reply:
I see. So DynamicConfigManager watches for entries in the zookeeper queue under `/config/changes/config_change_<sequence_number>`. That notification looks like this:

```
get /config/changes/config_change_0000000000
{"version":2,"entity_path":"topics/test-topic-2"}
```

So the simplest change would be to just have the controller add a notification to this queue for topic creation in addition to config alter.

I don't see a downside to doing that at the moment but I haven't tested it.

Reply:
Oh this is weirder than I thought. That znode only appears to be written to by AdminZkClient :(. I wonder how it works with newer APIs

Reply:
I pushed 92ba6bf which I'm not totally sure is correct yet. It does seem to work properly when testing locally so far. It basically just performs the same notification to that changes znode when the broker creates a topic configuration znode.

Reply:
My concern with doing this on the client side is that sometimes people use newer versions of open source clients without this path or obsolete client versions.
Thus I think it's safer to make this change on the server side, by having the controller to query the topic's configs during controller failover + when a new topic is created (besides using the dynamic config change mechanism).

Reply:
OK I pushed 071f7ed22 which I think handles this correctly.